### PR TITLE
#4342 re-fetch cleartext signed message when HTML version is corrupted

### LINKED ISSUE #4342: signed message occasionally throws an error 'Message digest did not match'
Description:

A user reported an issue where a signed message occasionally throws an error 'Message digest did not match'. I asked the responded to send a couple of signed messages and those worked well without any issue.

I looked closer into the signed message that throws 'Message digest did not match' by opening/sending it again with myself and it works well (needs modifying signerEmail not to trigger the missing pubkey error checks as its a different error with a different root cause).

I'm having trouble spotting the problem for this one as it works after moving opening it in my inbox but it shouldn't be the case. 

will try to move the .eml file to my inbox to dig into this issue.

### PR DESCRIPTION
**Pull request by rrrooommmaaa**
This PR re-fetches cleartext signed message from `text/plain` alternative part if signature verification fails

close #4342

issue #4810 (sets an example of how to test GmailElementReplacer)

issue #4929 (added a test helper method pgpBlockCheck for checking data in an existing pgp block)

----------------------------------

**Tests** _(delete all except exactly one)_:
- Tests added

--------------------------------

### To be filled by reviewers

I have reviewed that this PR... _(tick whichever items you personally focused on during this review)_:
- [x] addresses the issue it closes (if any)
- [x] code is readable and understandable
- [x] is accompanied with tests, or tests are not needed
- [x] is free of vulnerabilities
- [x] is documented clearly and usefully, or doesn't need documentation

### CODE REVIEW

File: test/source/tests/decrypt.ts [hunk 1]
```diff
@@ -892,6 +892,43 @@ XZ8r4OC6sguP/yozWlkG+7dDxsgKQVBENeG6Lw==
       })
     );
 
+    test(
+      'signature - cleartext signed messages from HTML are re-fetched when needed',
+      testWithBrowser('ci.tests.gmail', async (t, browser) => {
+        const acctEmail = 'ci.tests.gmail@flowcrypt.test';
+        const settingsPage = await browser.newExtensionSettingsPage(t, acctEmail);
+        const accessToken = await BrowserRecipe.getGoogleAccessToken(settingsPage, acctEmail); // todo: include in t?
```
Comment (rrrooommmaaa) [inline test/source/tests/decrypt.ts:900]: I think we can simplify some tests if we can extract google access token from `t` -- we can put it there in a helper method like `setUpCommonAcct`, so we don't have to open a dummy page just to retrieve the access token.

Comment (sosnovsky) [reply test/source/tests/decrypt.ts:900]: Yeah, it'll be nice to have access token available through `t`. Probably we can include such improvement when working on part 2 of #4494, there we can fetch access token when configuring test account.

**Review by sosnovsky — COMMENTED** [review]
Well done 👍  Just a small improvement for `eslint-disable`

File: test/source/tests/decrypt.ts (hunk 1 above)
Comment (sosnovsky) [reply test/source/tests/decrypt.ts:900]: #4966

**Review by sosnovsky — APPROVED** [review]
👍

**Pull request merged**